### PR TITLE
SignIn

### PR DESCRIPTION
**Pull request by Pavithrank02**
have changed signIN code check it

### CODE REVIEW

File: src/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    password: "",
+  });
+  const inputHanlder = event => {
+    const { name, value } = event.target;
+    setData((prev) => {
+      return { ...prev, [name]: value }
+    });
+  }
+
+  // useEffect(() => {
```
Comment (gowrishankkar) [inline src/SignIn.js:22]: remove comments

File: src/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { useState } from 'react'
+import { IconButton } from "@mui/material";
+import './SignIn.css'
+import AccountCircleRoundedIcon from "@mui/icons-material/AccountCircleRounded";
+import Home from './Home';
+import { useNavigate } from 'react-router-dom'
+
+function SignIn() {
+  const history = useNavigate()
+  const [data, setData] = useState({
```
Comment (gowrishankkar) [inline src/SignIn.js:11]: Use a separate object variable for default state

File: src/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  // useEffect(() => {
+  //     const inputVal = JSON.parse(localStorage.getItem('formValues'));
+  //     if (inputVal) {
+  //      setInputVal(inputVal);
+  //     }
+  //   }, []);
+  const submitButton = (e) => {
+    e.preventDefault();
```
Comment (gowrishankkar) [inline src/SignIn.js:29]: Remove prevent default

File: src/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    });
+  }
+
+  // useEffect(() => {
+  //     const inputVal = JSON.parse(localStorage.getItem('formValues'));
+  //     if (inputVal) {
+  //      setInputVal(inputVal);
+  //     }
+  //   }, []);
+  const submitButton = (e) => {
```
Comment (gowrishankkar) [inline src/SignIn.js:28]: submitButton  change to handleSubmit

File: src/SignIn.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    e.preventDefault();
+    const user = localStorage.getItem("formValues");
```
Comment (gowrishankkar) [inline src/SignIn.js:30]: use userDetails

**Review by gowrishankkar — CHANGES_REQUESTED** [review]

File: src/SignIn.js (hunk 4 above)
Comment (gowrishankkar) [inline src/SignIn.js:28]: Form should be able to handle the form information, no need to read it from state

**Review by gowrishankkar — CHANGES_REQUESTED** [review]

File: src/SignIn.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    // if(user.username === username && user.password === password){
+    //   console.log("match");
+    // } else {
+    //   console.log("doesnt match");
+    // }
+    const { username, password } = data;
+    if (username === "") {
+      alert("Please enter username")
+
+    } else if (password.length < 5) {
```
Comment (gowrishankkar) [inline src/SignIn.js:42]: use regex

File: src/SignIn.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      alert("Please enter valid password")
+    } else {
+      if (user && user.length) {
+        const userData = JSON.parse(user);
```
Comment (gowrishankkar) [inline src/SignIn.js:46]: find user data with find function

**Review by gowrishankkar — CHANGES_REQUESTED** [review]

File: src/SignIn.js (hunk 1 above)
Comment (Pavithrank02) [reply src/SignIn.js:22]: // const getUser = userData.filter((el, val) => {
        //   return el.username === username && el.password === password;
        // });
        // console.log(getUser)

File: src/SignIn.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        if (userData.username === username && userData.password === password) {
+          console.log("login successful");
+          history("/success")
+        } else if (userData.length === "") {
```
Comment (gowrishankkar) [inline src/SignIn.js:50]: const userFound = userDetails.find((userInfo) => userInfo.userName === name &&  userInfo.pwd === pwd )

if(userFound) //success
else // fail

**Review by gowrishankkar — CHANGES_REQUESTED** [review]

File: src/SignIn.js (hunk 1 above)
Comment (Pavithrank02) [reply src/SignIn.js:22]: done

File: src/SignIn.js (hunk 8 above)
Comment (Pavithrank02) [reply src/SignIn.js:50]: this logic isnt working

**Pull request closed without merging**